music21/analysis/harmonicFunction.py

```diff
@@ -0,0 +1,305 @@
+
+    >>> analysis.harmonicFunction.functionToRoman('Tp').figure
+    'vi'
+
+    >>> analysis.harmonicFunction.functionToRoman('Sg').figure
+    'vi'
+
+    The reverse operation is handled by the complementary
+    :func:`~music21.analysis.harmonicFunction.romanToFunction`.
+    In this case, :func:`~music21.analysis.harmonicFunction.romanToFunction`
+    follows the convention of preferring the P-version over alternatives.
+
+    >>> rn = roman.RomanNumeral('vi')
+    >>> str(analysis.harmonicFunction.romanToFunction(rn))
+    'Tp'
+
+    '''
+    if isinstance(keyOrScale, str):
+        keyOrScale = key.Key(keyOrScale)
+
+    referenceTuples = functionFigureTuplesMajor
+    if keyOrScale.mode == 'minor':
+        referenceTuples = functionFigureTuplesMinor
+
+    for thisKey, thisValue in referenceTuples.items():
+        if thisHarmonicFunction == thisKey:
+            return roman.RomanNumeral(thisValue, keyOrScale)
+
+    return False
+
+
+def romanToFunction(rn: roman.RomanNumeral,
+                    onlyHauptHarmonicFunction: bool = False
+                    ):
+    '''
+    Takes a Roman numeral and returns a corresponding harmonic function label.
+
+    >>> rn1 = roman.RomanNumeral('VI', 'a')
+    >>> fn1 = analysis.harmonicFunction.romanToFunction(rn1)
+    >>> fn1
+    <HarmonicFunction.SUBDOMINANT_MINOR_PARALLELKLANG_MAJOR>
+
+    This can be converted into a string:
+
+    >>> str(fn1)
+    'sP'
+
+    Optionally, set onlyHauptHarmonicFunction to True to return
+    a simplified version with only the HauptHarmonicFunction
+    (one of t, T, s, S, d, D: major and minor forms of the tonic, subdominant and dominant).
+
+    >>> fn1 = analysis.harmonicFunction.romanToFunction(rn1, onlyHauptHarmonicFunction=True)
+    >>> fn1
+    <HarmonicFunction.SUBDOMINANT_MINOR>
+
+    >>> str(fn1)
+    's'
+
+    Inversions are not currently considered (they may be in a future version of this).
+    This function simply uses the romanNumeral attribute of the roman.RomanNumeral object.
+    This excludes inversions, but
+    includes, where applicable, the frontAlterationAccidental.modifier.
+
+    >>> rn2 = roman.RomanNumeral('bII6', 'g')
+    >>> fn2 = analysis.harmonicFunction.romanToFunction(rn2)
+    >>> fn2
+    <HarmonicFunction.SUBDOMINANT_MINOR_GEGENKLANG_MAJOR>
+
+    >>> str(fn2)
+    'sG'
+
+    See further notes on the complementary
+    :func:`~music21.analysis.harmonicFunction.functionToRoman`.
+    '''
+
+    referenceTuples = functionFigureTuplesMajor
+    if rn.key:  # RomanNumeral object can be created without one.
+        if rn.key.mode == 'minor':
+            referenceTuples = functionFigureTuplesMinor
+
+    for thisKey, thisValue in referenceTuples.items():
+        if rn.romanNumeral == thisValue:
+            if onlyHauptHarmonicFunction:
+                return HarmonicFunction(str(thisKey)[0])
+            else:
+                return thisKey
+
+    return False
```

Comment:
sim.

Reply:
`None` = now done, but I don't immediately see how to do the / a more efficient version this way round.

Reply:
Oops, you're right, I didn't look closely. Sorry!